usr/share/lib/img_proof/tests/SLES/test_sles_hardened.py

```diff
@@ -77,9 +82,10 @@ def test_sles_hardened(host, get_release_value, is_sles_sap, is_sle_micro):
     )
 
     if result.rc != 0:
-        print("oscap command failed with {}".format(result.rc))
-        print("STDOUT: {}".format(result.stdout.strip()))
-        print("STDERR: {}".format(result.stderr.strip()))
+        with open(LOG_FILE, "w") as file, contextlib.redirect_stdout(file):
```

Comment:
Redirecting will likely cause trouble. Especially in threaded environment. This would be three levels of redirect nested. Also by default we don't really want to be dumping log files from within img-proof tests onto the host machine. Suggest this runs only if scap report is set and it writes to the file directly rather than doing a stdout redirect.

Reply:
Where are the three levels of redirection?

The way this redirect works is like a tee in Unix, writing to both the file and stdout.

We need to write to stdout so we can have a log from pytest as seen here:
https://openqa.suse.de/tests/13965566/file/img_proof-20240408084011.log

We also need to print it to a file so we can re-print the results on the last try as pytest seems to discard the stdout from previous tries.

Reply:
Pytest redirects certain output. We try to disable this but it doesn't seem to work for everything. img-proof redirects output during test runs to put it in the log file. So this would be a third location. And changing stdout affects other test runs if they are threaded. I.e. if you are running multiple tests at the same time there's a risk for a race condition.

Why we can't we dump everything to the img-proof log file? This would be a specific log file for the given test run.

Reply:
Fixed by explicitly printing to the log file and stdout.

We need this specific logfile only for the print in the last try.